packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1741,12 +1752,24 @@ class SemanticsConfiguration {
     _setFlag(SemanticsFlags.isSelected, value);
   }
 
+  /// Whether the owning [RenderObject] is currently enabled.
+  ///
+  /// A disabled object does not respond to user interactions. Only objects that
+  /// usually do respond to user interactions but currently don't (like a
+  /// disabled button) should be marked as disabled. Objects (like static text)
+  /// that never respond to user interactions should not be marked as disabled.
+  bool get isEnabled => _hasFlag(SemanticsFlags.hasEnabledState) ? _hasFlag(SemanticsFlags.isEnabled) : null;
+  set isEnabled(bool value) {
+    _setFlag(SemanticsFlags.hasEnabledState, true);
+    _setFlag(SemanticsFlags.isEnabled, value);
+  }
+
   /// If this node has Boolean state that can be controlled by the user, whether
   /// that state is on or off, corresponding to true and false, respectively.
   ///
   /// Do not set this to any value if the owning [RenderObject] doesn't have
   /// Booleans state that can be controlled by the user.
-  bool get isChecked => _hasFlag(SemanticsFlags.hasCheckedState) && _hasFlag(SemanticsFlags.isChecked);
+  bool get isChecked => _hasFlag(SemanticsFlags.hasCheckedState) ? _hasFlag(SemanticsFlags.isChecked) : null;
   set isChecked(bool value) {
```

Comment:
Can we add an `assert` to uphold the "Do not set this to any value if the owning [RenderObject] doesn't have Booleans state that can be controlled by the user" contract?

Reply:
Unfortunately, we can't detect that. This is where the RenderObject defines that it has checked/unchecked state.

Reply:
Ah, nvm then.